src/cpp/common/alarm.cc

```diff
@@ -31,21 +31,17 @@
  */
 
 #include <grpc++/alarm.h>
-#include <grpc++/completion_queue.h>
 #include <grpc++/impl/grpc_library.h>
-#include <grpc/grpc.h>
 
 namespace grpc {
 
 static internal::GrpcLibraryInitializer g_gli_initializer;
-Alarm::Alarm(CompletionQueue* cq, gpr_timespec deadline, void* tag)
-    : tag_(tag),
-      alarm_(grpc_alarm_create(cq->cq(), deadline, static_cast<void*>(&tag_))) {
+
+Alarm::~Alarm() {
   g_gli_initializer.summon();
```

Comment:
Because it can't live in the constructor, now in a .h due to the use of templates. The static global variable needs to live inside a compilation unit.
You make a good point about the CompletionQueue, hmmm. I guess we could remove it provided we put a big comment in its stead pointing at how if the CQ is ever removed, it'd have to be added back.